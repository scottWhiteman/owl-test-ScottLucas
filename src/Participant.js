import React from 'react';

class Participant extends React.Component {
  isInSession(inSession, onStage) {
    if (onStage) {
      return <p>on stage</p>;
    } else if (inSession) {
      return <p>in session</p>
    }
    return <p>left session</p>;
  }
  
  render() {
    return (
      <li className="Participant" id={this.props.id}>
        <img src={this.props.avatar} alt="this user's avatar"/>
        <div className="UserStatus">
          <p>{this.props.name}</p>
          {this.isInSession(this.props.inSession, this.props.onStage)}
        </div>
      </li>
    )
  }
}

export default Participant;