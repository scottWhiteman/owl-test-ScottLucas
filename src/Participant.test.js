import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Participant from './Participant';

it ("Participants render without crash", () => {
    const div = document.createElement("div");
    ReactDOM.render(
        <Participant 
            id={800}
            name={"Joey"}
            avatar={"https://robohash.org/doloredolorescupiditate.jpg?size=200x200&set=set1"}
            inSession={false}
            onStage={false}
        />
    , div);
    ReactDOM.unmountComponentAtNode(div);
});

it ("Renders Participant to UI", () => {
    const tree = renderer.create(<Participant 
        id={800}
        name={"Joey"}
        avatar={"https://robohash.org/doloredolorescupiditate.jpg?size=200x200&set=set1"}
        inSession={false}
        onStage={false}
    />).toJSON();
    expect(tree).toMatchSnapshot();
});