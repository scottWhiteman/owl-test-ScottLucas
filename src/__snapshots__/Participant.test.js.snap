// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Participant to UI 1`] = `
<li
  className="Participant"
  id={800}
>
  <img
    alt="this user's avatar"
    src="https://robohash.org/doloredolorescupiditate.jpg?size=200x200&set=set1"
  />
  <div
    className="UserStatus"
  >
    <p>
      Joey
    </p>
    <p>
      left session
    </p>
  </div>
</li>
`;
